pkg/kubelet/dockertools/docker.go

```diff
@@ -348,13 +360,40 @@ func milliCPUToShares(milliCPU int64) int64 {
 
 // GetKubeletDockerContainers lists all container or just the running ones.
 // Returns a map of docker containers that we manage, keyed by container ID.
-// TODO: Move this function with dockerCache to DockerManager.
+// TODO: Replace this function with GetKubeletManagedContainers directly.
 func GetKubeletDockerContainers(client DockerInterface, allContainers bool) (DockerContainers, error) {
-	result := make(DockerContainers)
-	containers, err := client.ListContainers(docker.ListContainersOptions{All: allContainers})
-	if err != nil {
-		return nil, err
+	if allContainers {
+		return GetKubeletManagedContainers(client, listAll)
+	} else {
+		return GetKubeletManagedContainers(client, listRunning)
+	}
+}
+
+type listContainersType string
+
+const (
+	listAll     listContainersType = "all"
+	listRunning listContainersType = "running"
+	listExited  listContainersType = "exited"
+)
+
+func getContainers(client DockerInterface, ctype listContainersType) ([]docker.APIContainers, error) {
+	switch ctype {
+	case listAll:
+		return client.ListContainers(docker.ListContainersOptions{All: true})
+	case listRunning:
+		return client.ListContainers(docker.ListContainersOptions{All: false})
+	case listExited:
+		return client.ListContainers(docker.ListContainersOptions{
+			All:     true,
+			Filters: map[string][]string{"status": {DockerExited}}})
+	default:
+		return nil, fmt.Errorf("unknown container type: %q", ctype)
```

Comment:
Is this really a container type?  It's the type of subset, or the 'list option', but container type seems incorrect.